### PR TITLE
Issue #120: Updated ldavis.js to be compatible with latest d3.v5

### PR DESCRIPTION
**Pull request by dtemkin**
CHANGED 
  scale.linear(...)
TO
  scaleLinear(...)

CHANGED
    .axis(...).scale(...).orient(...) 
TO
    axisLeft (_y_)(...), axisTop(...) (_x_) or axisBottom(...) (_scaleX_) 

CHANGED
    d3.scaleordinal(...).domain(...).rangeRoundBands(...)
TO
    d3.scaleBand().domain(...).rangeBands(...).padding(...)

CHANGED
    y.rangeBand(...) 
TO
    y.bandwidth(...)

REPLACED d3.v3.js with d3.v5.js

ADDED:
    optional function parameters that let user specify primary (color1) and 
secondary (color2) colors. (Default colors are originals)

This should also be compatible with d3.v4 but I havent tested it yet.

### CODE REVIEW

**Comment by dtemkin** [comment]
I looked at the errors and they appear to be the result of an issue in pandas. I can attempt to address them if necessary to merge my PR.

**Review by bmabey — APPROVED** [review]

**Comment by bmabey** [comment]
Could you resolve the conflicts in ldavis.js?

**Comment by msusol** [comment]
Are these changes still relevant? I can make an attempt to resolve the conflicts in ldavis.js

Is there any other effort to upgrade d3.v3 to d3.v5+ ?

**Comment by dtemkin** [comment]
@msusol @bmabey Sorry this took me so long to complete. I fixed the references to d3 so that they are specifically referencing d3.v5.min.js or https://d3js.org/d3.v5.js . I believe everything is good to go, but take a look when you get a chance to make sure I am not missing something. 

I also fixed the problems with the travis-ci configuration in which I had to separate the dependencies for python 3.5.

File: pyLDAvis/_prepare.py [hunk 1]
```diff
@@ -188,12 +188,11 @@ def _series_with_name(data, name):
         return pd.Series(data, name=name)
 
 
-def _topic_coordinates(mds, topic_term_dists, topic_proportion, start_index=1):
```
Comment (bmabey) [inline pyLDAvis/_prepare.py:191]: Is there a reason why you changed this line?

File: pyLDAvis/_prepare.py [hunk 2]
```diff
@@ -188,12 +188,11 @@ def _series_with_name(data, name):
...
+def _topic_coordinates(mds, topic_term_dists, topic_proportion):
     K = topic_term_dists.shape[0]
     mds_res = mds(topic_term_dists)
     assert mds_res.shape == (K, 2)
-    mds_df = pd.DataFrame({'x': mds_res[:, 0], 'y': mds_res[:, 1],
-                           'topics': range(start_index, K + start_index),
```
Comment (bmabey) [inline pyLDAvis/_prepare.py:196]: Ditto here.. my guess is that this is an accident...

File: pyLDAvis/_prepare.py [hunk 3]
```diff
@@ -217,24 +216,22 @@ def _job_chunks(l, n_jobs):
 
 def _find_relevance(log_ttd, log_lift, R, lambda_):
     relevance = lambda_ * log_ttd + (1 - lambda_) * log_lift
-    return relevance.T.apply(lambda topic: topic.nlargest(R).index)
+    return relevance.T.apply(lambda s: s.sort_values(ascending=False).index).head(R)
```
Comment (bmabey) [inline pyLDAvis/_prepare.py:219]: ditto

File: pyLDAvis/_prepare.py [hunk 4]
```diff
@@ -217,24 +216,22 @@ def _job_chunks(l, n_jobs):
...
 
 
 def _find_relevance_chunks(log_ttd, log_lift, R, lambda_seq):
     return pd.concat([_find_relevance(log_ttd, log_lift, R, l) for l in lambda_seq])
 
 
 def _topic_info(topic_term_dists, topic_proportion, term_frequency, term_topic_freq,
-                vocab, lambda_step, R, n_jobs, start_index=1):
```
Comment (bmabey) [inline pyLDAvis/_prepare.py:228]: ditto, I don't see why these changes are needed for this PR

**Review by bmabey — CHANGES_REQUESTED** [review]
It looks like you maybe forgot to rebase to get the latest changes? Python code changes are happening that seem unrelated to updating to the latest d3.

**Comment by bmabey** [comment]
LGTM, thank you for this @dtemkin!

**Comment by msusol** [comment]
Thank you!!

**Comment by msusol** [comment]
Thank you, Dan!

Ben, when will there be a new version tag available?

**Comment by msusol** [comment]
Dan,

Please see the recent issues about missing the labels in the d3.v5
rendering.

**Comment by msusol** [comment]
See follow up refactoring: #198

**Pull request merged**